Trials.com/react-frontend/src/components/submit-run.jsx

```diff
@@ -0,0 +1,227 @@
+          step={0.5}
+          min={0.5}
+          max={8.5}
+          valueLabelDisplay="auto"
+          marks={marks}
+          name="ninjaLevel"
+          onChange={(event, value) => formik.setFieldValue("ninjaLevel", value)}
+        />
+        <label>Faults</label>
+        <input
+          placeholder="241"
+          id="faults"
+          name="faults"
+          type="number"
+          onKeyDown={(evt) =>
+            (evt.key === "e" ||
+              evt.key === "E" ||
+              evt.key === "-" ||
+              evt.key === "+" ||
+              evt.key === ".") &&
+            evt.preventDefault()
+          }
+          onChange={(e) => {
+            e.preventDefault();
+            const { value } = e.target;
+            if (value < 500) {
+              formik.setFieldValue("faults", value);
+            }
+          }}
+          value={formik.values.faults}
+        />
+        <label>Time</label>
+
+        <InputMask
+          placeholder="12:33.677"
+          mask="99:99:999"
+          name="time"
+          type="text"
+          onChange={formik.handleChange}
+          value={formik.values.time}
+        ></InputMask>
+        <label>Length</label>
+        <select
+          name="length"
+          type="text"
+          onChange={formik.handleChange}
+          value={formik.values.length}
+        >
+          <option value="" label="Select Length of track" />
+          <option value="Short" label="Short" />
+          <option value="Medium" label="Medium" />
+          <option value="Long" label="Long" />
+        </select>
+        <label>Fault Sponginess</label>
+        <select
+          name="faultSponginess"
+          type="text"
+          onChange={formik.handleChange}
+          value={formik.values.faultSponginess}
+        >
+          <option value="" label="Select Fault Sponginess of track" />
+          <option value="Not At All" label="Not At All" />
+          <option value="Not Very" label="Not Very" />
+          <option value="Moderately" label="Moderately" />
+          <option value="Very" label="Very" />
+          <option value="Extremely" label="Extremely" />
+        </select>
+        <label>How did you like the track? (optional)</label>
+        <ReactStars {...Stars} onChange={setRating} />
+        <button type="submit">Submit</button>
+      </form>
+    </Formik>
+  );
+};
+
+const marks = [
+  {
+    value: 1,
+    label: "1",
+  },
+
+  {
+    value: 2,
+    label: "2",
+  },
+
+  {
+    value: 3,
+    label: "3",
+  },
+
+  {
+    value: 4,
+    label: "4",
+  },
+
+  {
+    value: 5,
+    label: "5",
+  },
+
+  {
+    value: 6,
+    label: "6",
+  },
+
+  {
+    value: 7,
+    label: "7",
+  },
+
+  {
+    value: 8,
+    label: "8",
+  },
+
+  {
+    value: 9,
+  },
+];
+
+const Stars = {
+  size: 50,
+  value: 0,
+  isHalf: true,
+  a11y: true,
+  activeColor: "blue",
+};
+
+function valuetext(value) {
+  return `${value}`;
+}
+
+SubmitRun.propTypes = {
```

Comment:
i was thinking about this a bit and im not sure how useful props used like this will be? i dont think it hurts, but the props aren't actually being used anywhere in the file from what i see. the way i see it is props will be something we can use to slightly change how some components might look by putting it in our code. here you are kinda using it like a restriction for the form, but thats already being taken care of in the form itself as seen in the formik. as an example, say we were to reuse this submit panel somewhere else but we wanted a different title, you would have title be a prop with type string. then instead of just putting "Submit your run" like you did in an h3, it'd be props.title. that way when its reused but altered easily. unless im missing something, thats how i see props and proptypes as. its for our code so we know when we are reusing a component somewhere else, which props exist and what type they should be.

Reply:
it wouldnt hurt, but like you said i also dont know when this would really be reused differently. and it wouldnt be hard to add props in the future either if we wanted to, so id say its fine for now.